forgerock-openbanking-backstage/forgerock-openbanking-metrics/forgerock-openbanking-metrics-services/src/main/java/com/forgerock/openbanking/analytics/ForgerockOpenbankingMetricsServicesApplication.java

```diff
@@ -91,194 +93,216 @@
-			//You can load more GrantedAuthority based on the user subject, like loading the TPP details from the software ID
-			return authentication;
-		}
-
-		@Override
-		public boolean supports(Class<?> aClass) {
-			return true;
-		}
-	}
-
-	@Bean
-	public RestTemplate restTemplate(@Qualifier("mappingJacksonHttpMessageConverter")
-											 MappingJackson2HttpMessageConverter converter) throws SslConfigurationFailure {
-		RestTemplate restTemplate = new RestTemplate(sslConfiguration.factory(keyAlias, true));
-		customiseRestTemplate(converter, restTemplate);
-		return restTemplate;
-	}
-
-	@Bean
-	public WebClient webClient() throws Exception {
-
-		SslContext sslContext = sslConfiguration.getSslContextForReactor(keyAlias);
-		HttpClient httpClient = HttpClient.create()
-				.secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
-		ReactorClientHttpConnector connector = new ReactorClientHttpConnector(httpClient);
-		return WebClient.builder().clientConnector(connector).build();
-	}
-
-	private void customiseRestTemplate(@Qualifier("mappingJacksonHttpMessageConverter") MappingJackson2HttpMessageConverter converter, RestTemplate restTemplate) {
-		List<HttpMessageConverter<?>> messageConverters = restTemplate.getMessageConverters();
-		messageConverters.removeIf(c -> c instanceof MappingJackson2HttpMessageConverter);
-		messageConverters.add(converter);
-		restTemplate.setErrorHandler(new ClientResponseErrorHandler());
-	}
-
-	@Bean
-	public MappingJackson2HttpMessageConverter mappingJacksonHttpMessageConverter(@Qualifier("objectMapperBuilderCustomizer") Jackson2ObjectMapperBuilderCustomizer objectMapperBuilderCustomizer) {
-		MappingJackson2HttpMessageConverter converter = new MappingJackson2HttpMessageConverter();
-		Jackson2ObjectMapperBuilder objectMapperBuilder = new Jackson2ObjectMapperBuilder();
-		objectMapperBuilderCustomizer.customize(objectMapperBuilder);
-		converter.setObjectMapper(objectMapperBuilder.build());
-		return converter;
-	}
-
-	public enum AnalyticsAuthority implements GrantedAuthority {
-		PUSH_KPI,
-		READ_KPI;
-
-		@Override
-		public String getAuthority() {
-			return name();
-		}
-	}
+            super(
+                    "JWTCookie",
+                    tokenSerialised -> {
+                        try {
+                            return cryptoApiClient.decryptJwe(tokenSerialised);
+                        } catch (JOSEException e) {
+                            throw new RuntimeException(e);
+                        }
+                    },
+                    token -> token.getJWTClaimsSet().getSubject(),
+                    authoritiesCollector,
+                    cookieName
+            );
+        }
+    }
+
+    /*
+     * Adding the default authorities.
+     * Adding the authorities coming from Forgerock AM JWT Cookie authentication if exist authorities.
+     * FYI: Additional Authorities setted on Claim 'group' set in 'identity / MSISDN Number'.
```

Comment:
I think it would be good to add a bit of context here to explain why this is needed from a business point of view.

Reply:
Comment added to clarify the context of JwtCookieAuthorityCollector resposabilyties